### ISSUE TITLE
Adding kwargs hides incompatible method override error

### ISSUE THREAD

**Issue by Daraan** [microsoft/pyright#10942]
**Describe the bug**
👋 See the snippet below, `Child.bar` with kwargs does not raise an error, while `Child.foo` does. It feels strange that only one of them raises an error.

**Code or Screenshots**

Code sample in [pyright playground](https://pyright-play.net/?code=MYGwhgzhAEBCkFMBcBYAUO6XoBMEDNp8B7YgCggRHwBpowlcBLYAFwG0JWAnOr7gLoBKRgDpx6TNjyEARmG4UqteoxwsO-Pj2FiJGNKEgwAogA9WZeJRGS02XASKkl1Og2ZtOPOsVkArBDZdaHFROwcZaHlFSjdVT00faD9A4JFQ-XQjKGgAYQALJhAcMnNWOmsEWwNIpxJyOJUwDLCI6ScY12a6ACpegGsAdwUAcwhW-SA)

```python
class Base:

    def foo(self, a: dict[str, str]): ...

    def bar(self, a: dict[str, str]): ...

class Ext(Base):

    def foo(self, a: dict[str, object]): ...

    def bar(self, a: dict[str, object]): ...

class Child(Ext, Base):

    def foo(self, a): ...  # Error

    def bar(self, a, **kwargs): ...  # OK

```

**VS Code extension or command-line**
1.1.405

**Comment by erictraut** [comment]
This is related to parameter type inference for inherited methods. As you pointed out in a [recent enhancement request](https://github.com/microsoft/pyright/issues/10952), pyright infers the types of method parameters in the specific case where the method is unannotated, its parent class has an annotated method of the same name, and the parameter count and names match. I added this inference behavior to improve the user experience in some language server scenarios, but in retrospect I kind of regret adding it. It leads to hard-to-explain behaviors like the one you're pointing out here. I'm half tempted to remove the feature, but I suspect that I'd get quite a few complaints if I did.

In any case, pyright is working as designed here, so I don't consider this a bug.
